batch/batch/server/database.py

```diff
@@ -23,20 +23,43 @@ def __init__(self, db):
         super().__init__(db, 'jobs')
 
     async def update_record(self, batch_id, job_id, **items):
+        assert 'cancelled' not in items
```

Comment:
Should we handle this?

Reply:
In what sense?

Reply:
these asserts raise an error if false, that error isn't caught. What does the user get, a 500, or does the process just crash?

Reply:
Also, if "cancelled" is in items, and we properly handle data races, the returned error should be >= 400 <= 500, to indicate that this is a client error.

Reply:
cancelled being in items is a user error of the JobsTable. This has nothing to do with the client. We have lots of asserts in our code that are developer assertions. I think this is fine as is.

Reply:
Right, since this is a user error the client (uset) should be informed it’s a 400-order error

Reply:
A developer assertion is used for different reasons than user-facing assertions. If the goal is to show programming error, then an assert and say a 500 return to the user is appropriate. 400 for user errors